### PR TITLE
Riese/multi select case tile

### PR DESCRIPTION
**Pull request by MartinRiese**
## Product Description
Add support for multi select on the case tile list in WebApps

## Technical Summary

This PR already includes changes from https://github.com/dimagi/commcare-hq/pull/32933. To see the diff between the two branches click [here](https://github.com/dimagi/commcare-hq/compare/em/dynamic-case-tile-widths...riese/multi_select_case_tile).

https://dimagi-dev.atlassian.net/browse/USH-3136

Only the normal case list had support for multi select so far. This PR adds it for case list tiles.

## Feature Flag
case_list_tile, with some options under case_list_custom_xml

## Safety Assurance

Will only affect those using case tiles in web apps (very few if any at the moment), and only changes the UI visually. Does not change how data is loaded into the tiles or how they are interacted with to select cases.

### Safety story

Tested locally and will test in staging

### Automated test coverage

Some ui test had to be updated due to changes in the layout

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by orangejenny** [comment]
(ignoring for now because the commits are mixed up with Evan's PR, but do ping when you want review)

**Comment by nospame** [comment]
Echoing @orangejenny's comment above, I do think this should contain just the changes related to multi-select if at all possible.

**Review by orangejenny — APPROVED** [review]
How thoroughly did you test the existing (non-tile) multiselect functionality?

**Comment by MartinRiese** [comment]
@nospame Just saw you merged your PR into master. So now this is a clean diff

File: corehq/apps/cloudcare/templates/formplayer/case_list.html [hunk 1]
```diff
@@ -17,17 +17,22 @@
...
     <div class="module-case-list-table-container">
       <table class="table module-table module-table-case-list">
-        <% if(!useTiles) { %>
+
         <thead>
         <tr>
           <% if (isMultiSelect && !hasNoItems) { %>
           <th class="module-case-list-header"><input type="checkbox" id="select-all-checkbox"/></th>
           <% } %>
+          <% if(!useTiles) { %>
```
Comment (nospame) [inline corehq/apps/cloudcare/templates/formplayer/case_list.html:35]: What is this check doing? It seems like it's already determined `useTiles` is false given this is in the `else` from line 26.

Comment (nospame) [reply corehq/apps/cloudcare/templates/formplayer/case_list.html:35]: Oh, or maybe that `else` is actually the alternative to `<% if (isMultiSelect && !hasNoItems) { %>` from line 21, and the whole thing is within line 20's `<% if (useTiles) { %>` ? I think there's maybe some indentation that could be cleared up in this section.

Comment (MartinRiese) [reply corehq/apps/cloudcare/templates/formplayer/case_list.html:35]: No, I think you are right. This cannot be false at this point

**Review by nospame — APPROVED** [review]

**Pull request merged**